lib/factories/createStrategyFactory.js

```diff
@@ -0,0 +1,133 @@
+                value: false
+              }
+            ])
+          .then(function (userChoice) {
+            return userChoice.value;
+          });
+      },
+      getBusinessGroup: function (businessGroups) {
+        return commandPrompt.getChoice(messages.businessGroupPromptMessage(),
+          'name', 'id', sortByName(businessGroups));
+      },
+      getAPI: function (apis) {
+        return commandPrompt.getChoice(messages.apiPromptMessage(),
+          'name', 'id', sortByName(apis));
+      },
+      getAPIName: function (apis) {
+        return commandPrompt.getInput(messages.apiNamePromptMessage())
+          .then(function (apiName) {
+            return checkAPINameIsNotUsed(apis, apiName)
+              .return(apiName);
+          });
+      },
+      getAPIVersionName: function (apis, apiId) {
+        return commandPrompt.getInput(messages.apiVersionNamePromptMessage())
+          .then(function (versionName) {
+            return checkAPIVersionNameIsNotUsed(apis, apiId, versionName)
+              .return(versionName);
+          });
+      },
+      getRootRamlPath: function (filesPath) {
+        return commandPrompt.getRawChoice(messages.rootRamlPathPromptMessage(),
+          filesPath.sort());
+      }
+    };
+  }
+
+  function batchStrategy(parameters) {
+    return {
+      getCreateAPIorAPIVersionChoice: function () {
+        return BPromise.resolve(!!parameters.apiName);
+      },
+      getBusinessGroup: function (businessGroups) {
+        var businessGroup = _.find(businessGroups, 'id', parameters.bizGroup);
+        return businessGroup ?
+          BPromise.resolve(businessGroup) :
+          BPromise.reject(new errors.ChoiceNotFoundError(
+            messages.businessGroupDescription()));
+      },
+      getAPI: function (apis) {
+        var api = _.find(apis, 'id', parameters.apiId);
+        return api ?
+          BPromise.resolve(api) :
+          BPromise.reject(new errors.ChoiceNotFoundError(
+            messages.apiDescription()));
+      },
+      getAPIName: function (apis) {
+        return checkAPINameIsNotUsed(apis, parameters.apiName)
+          .return(parameters.apiName);
+      },
+      getAPIVersionName: function (apis, apiId) {
+        return checkAPIVersionNameIsNotUsed(apis, apiId, parameters.apiVersion)
+          .return(parameters.apiVersion);
+      },
+      getRootRamlPath: function (filesPath) {
+        var localRootRamlPath = '/' + parameters.rootRamlPath;
+        var rootRamlExists = _.includes(filesPath, localRootRamlPath);
+        return rootRamlExists ?
+          BPromise.resolve(localRootRamlPath) :
+          BPromise.reject(new errors.ChoiceNotFoundError(
+            messages.rootRamlDescription()));
+      }
+    };
+  }
+
+  function checkAPINameIsNotUsed(apis, apiName) {
+    var existingAPI = _.find(apis, 'name', apiName.toString());
+    if (existingAPI) {
+      return BPromise.reject(new errors.RepeatedAPINameError(
```

Comment:
it seems a little strong to make the command fail if the name is repeated. I think a better solution would be to catch the error and prompt the user for a different name